cpp/new_journal_alert.cc

```diff
@@ -172,6 +206,220 @@ namespace std {
 namespace {
 
 
+class GenerateEmailContents {
+public:
+    ~GenerateEmailContents() {};
+    virtual std::string generateContent(const std::string &user_type, const std::string &name_of_user, const std::string &vufind_host,
```

Comment:
Seems like each of the two derived classes implements only one of these methods and aborts in the other one with a "not implemented" message. But i still think it's a good idea to implement a class concept.

One possibility might be to merge the two functions by changing the signature of generateContent to take a superset of all provided information (e.g. as a struct), and then each function only uses the subset of information that it actually needs.

What do you think?

Reply:
As discussed.